client/post-editor/media-modal/test/index.jsx

```diff
@@ -85,194 +76,229 @@ describe( 'EditorMediaModal', () => {
 	} );
 
 	test( 'When `single` selection screen chosen should initialise with no items selected', () => {
-		shallow( <EditorMediaModal { ...baseProps } single={ true } view={ null } /> ).instance();
+		const { container } = render(
+			<EditorMediaModal { ...baseProps } single={ true } view={ null } />
+		);
+		Modal.setAppElement( container );
 		expect( selectMediaItems ).toHaveBeenCalledWith( DUMMY_SITE.ID, [] );
 	} );
 
-	test( 'should prompt to delete a single item from the list view', () => {
-		const media = DUMMY_MEDIA.slice( 0, 1 );
+	test( 'should prompt to delete a single item from the list view', async () => {
+		const user = userEvent.setup();
 
-		const tree = shallow(
-			<EditorMediaModal { ...baseProps } selectedItems={ media } />
-		).instance();
-		tree.deleteMedia();
-
-		return new Promise( ( resolve ) => {
-			process.nextTick( function () {
-				expect( deleteMedia ).toHaveBeenCalledWith(
-					DUMMY_SITE.ID,
-					media.map( ( { ID } ) => ID )
-				);
-				resolve();
-			} );
+		MediaLibrary.mockImplementationOnce( ( { onDeleteItem } ) => {
+			return <button onClick={ onDeleteItem }>Delete</button>;
 		} );
+
+		const media = DUMMY_MEDIA.slice( 0, 1 );
+		const props = { ...baseProps, selectedItems: media };
+		const { container } = render( <EditorMediaModal { ...props } /> );
+
+		Modal.setAppElement( container );
+
+		const btn = screen.getByRole( 'button', { name: 'Delete' } );
```

Comment:
Feel free to remove the constant and directly inline the screen queries if they're used only once.